context_builder.Dockerfile

```diff
@@ -2,21 +2,32 @@
 # for creating docker build context out of a Truss.
 # Build that image as:
 # docker buildx build . -f context_builder.Dockerfile --platform=linux/amd64 -t baseten/truss-context-builder
-FROM python:3.9-slim
+FROM python:3.9-slim AS builder
 
 RUN apt-get update \
-    && apt-get install --yes --no-install-recommends curl tar \
+    && apt-get install --yes --no-install-recommends curl build-essential golang-go \
```

Comment:
We add the tooling needing for arm64 images unconditionally, but since it's a multi stage image, these deps aren't needed in the final image